src/mame/drivers/scopus.cpp

```diff
@@ -0,0 +1,233 @@
+// license:GPL2+
+// copyright-holders:Felipe Sanches
+/********************************************************************
+
+   Scopus Sagitta 150 / 180
+   Serial terminal
+
+   driver by: Felipe Correa da Silva Sanches <juca@members.fsf.org>
+
+   Thanks to Werner Moecke for the help in inspecting the circuits and
+     debugging this driver.
+
+   TODO:
+   - figure out how the additional ROMs are mapped into memory
+   - figure out how/if the ROMs of the two models (150 and 180) can
+     be switched (is that via a dipswitch setting?) The model 180
+     contains a Model 150 ROM as well.
+   - emulate the printer port
+   - document the keyboard layout and emulate it
+
+*********************************************************************/
+
+#include "emu.h"
+#include "cpu/i8085/i8085.h"
+#include "machine/i8251.h"
+#include "machine/i8257.h"
+#include "machine/i8212.h"
+#include "video/i8275.h"
+#include "screen.h"
+#include "bus/rs232/rs232.h"
+#include "machine/clock.h"
+
+class sagitta180_state : public driver_device
+{
+public:
+	sagitta180_state(const machine_config &mconfig, device_type type, const char *tag)
+		: driver_device(mconfig, type, tag),
+		m_palette(*this, "palette"),
+		m_crtc(*this, "crtc"),
+		m_dma8257(*this, "dma"),
+		m_maincpu(*this, "maincpu"){ }
+
+	/* devices */
+        required_device<palette_device> m_palette;
+        required_device<i8275_device> m_crtc;
+	required_device<i8257_device> m_dma8257;
+	required_device<cpu_device> m_maincpu;
+
+	DECLARE_DRIVER_INIT(sagitta180);
+	DECLARE_WRITE_LINE_MEMBER(hrq_w);
+	DECLARE_READ8_MEMBER(memory_read_byte);
+        I8275_DRAW_CHARACTER_MEMBER(crtc_display_pixels);
+
+
+	// Character generator
+	const uint8_t *m_chargen;
```

Comment:
Please use a `required_region_ptr` for this so some classes of error can be caught when running `-validate` rather than causing null pointer errors when running (see e.g. src/mame/drivers/alphatro.cpp for an example).